mdtraj/core/trajectory.py

```diff
@@ -369,69 +369,126 @@ def load(filename_or_filenames, discard_overlapping_frames=False, **kwargs):
     except KeyError:
         raise IOError('Sorry, no loader for filename=%s (extension=%s) '
-                      'was found. I can only load files '
-                      'with extensions in %s' % (filename, extension, FormatRegistry.loaders.keys()))
+                    'was found. I can only load files '
+                    'with extensions in %s' % (
+                        filename_or_filenames[0], extension, FormatRegistry.loaders.keys()))
+
+    if loader.__name__ not in ['load_dtr']:
+            _assert_files_exist(filename_or_filenames)
+    else:
+        _assert_files_or_dirs_exist(filename_or_filenames)
+
+
+    if extension not in _TOPOLOGY_EXTS:
+        # standard_names is a valid keyword argument only for files containing topologies
+        kwargs.pop('standard_names', None)
 
-    if extension in _TOPOLOGY_EXTS:
+    trajectories = []
+    tmp_file = filename_or_filenames.pop(0)
+    try:
         # this is a little hack that makes calling load() more predictable. since
         # most of the loaders take a kwargs "top" except for load_hdf5, (since
         # it saves the topology inside the file), we often end up calling
         # load_hdf5 via this function with the top kwarg specified. but then
         # there would be a signature binding error. it's easier just to ignore
         # it.
-        if 'top' in kwargs:
-            warnings.warn('top= kwarg ignored since file contains topology information')
-            kwargs.pop('top', None)
-    else:
-        # standard_names is a valid keyword argument only for files containing topologies
-        kwargs.pop('standard_names', None)
+        #TODO make all the loaders accept a pre parsed topology (top) in order to avoid
+        #this part and have a more consistent interface and a faster load function
+        t = loader(tmp_file, **kwargs)
+        
+    except TypeError as e:
 
-    if loader.__name__ not in ['load_dtr']:
-        _assert_files_exist(filename_or_filenames)
-    else:
-        _assert_files_or_dirs_exist(filename_or_filenames)
+        #Don't want to intercept legit
+        #TypeErrors
+        if "got an unexpected keyword argument 'top'" not in str(e):
+            raise
+
+        warnings.warn('top= kwargs ignored since this file parser does not support it')
+
+        kwargs.pop('top', None)
+
+        t = loader(tmp_file, **kwargs)
+
+    except ValueError as e:
+
+        if 'xyz must be shape' in str(e):
+
+            raise ValueError('The topology and the trajectory files might not contain the same atoms\n'
+            'The input topology must contain all atoms even if '
+            'you want to select a subset of them with atom_indices'
+            ) from e
+
+        raise
+
+    trajectories.append(t)
+
+    # Only do this monkey patching if needed in order not to
+    # modify the output topology
+    if ('top' in kwargs) and (
+        kwargs.get('atom_indices', None) is not None) and (
+        len(filename_or_filenames) > 1):
```

Comment:
`top` is always in kwargs (set on line 396), and the `len` should check for `> 0` after the `pop` on `419`
```suggestion
    if kwargs.get('atom_indices', None) is not None) and (
        len(filename_or_filenames) > 0):
```